74.py

```diff
@@ -0,0 +1,31 @@
+class Solution:
+    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
+        # outer binary search to find correct row
+        row_index = -1
+        left, right = 0, len(matrix) - 1
+        while left <= right:
+            i = left + ((right - left) // 2)
+            value = matrix[i][0]
+            if value == target:
+                return True
+            elif value < target:
+                if matrix[i][-1] < target:
+                    left = i + 1
+                else:
+                    row_index = i
+                    break
+            elif value > target:
+                right = i - 1
+        # inner binary search
+        row = matrix[row_index]
+        left, right = 0, len(row) - 1
+        while left <= right:
+            i = left + ((right - left) // 2)
```

Comment:
Typically, this variable is called mid = left + ((right - left) // 2) Please make this change and keep this in mind for all future binary search.